internal/api/external/oidc/oidc.go

```diff
@@ -190,15 +191,32 @@ func GetUser(ctx context.Context, code string, state string) (User, error) {
 		ClientID: clientID,
 	}
 
-	idToken, err := auth.provider.Verifier(oidcConfig).Verify(ctx, rawIDToken)
-	if err != nil {
+	if _, err := auth.provider.Verifier(oidcConfig).Verify(ctx, rawIDToken); err != nil {
 		return User{}, errors.Wrap(err, "verify id token error")
 	}
 
 	var user User
-	if err := idToken.Claims(&user); err != nil {
+
+	// Request the claims through a UserInfo call to the OIDC server.
+	// We don't read claims from the "id_token" because most OIDC servers don't include
+	// claims in id_token by default. We would have to request claims to be included in
+	// id_token explicitly. But then servers don't have to implement/support that
+	// request. The UserInfo call does always include the claims.
+	userInfo, err := auth.provider.UserInfo(ctx, oauth2.StaticTokenSource(token))
+	if err != nil {
 		return User{}, errors.Wrap(err, "get userInfo error")
 	}
 
+	// Parse the well-known claims into user.
+	if err := userInfo.Claims(&user); err != nil {
```

Comment:
I am trying to understand the syntax of GO code for my own learning, does this mean that the contents of the userInfo.Claims object will be used to populate the User object?

Reply:
Claims is a method. the comment above the code says the claims are parsed into user indeed.